### PR TITLE
UI: Change layout

### LINKED ISSUE #57: Update logo to work for both light and dark themes

### LINKED ISSUE #58: Unintended dark line at the top of the command box under some screen resolution settings
ttt

### PR DESCRIPTION
**Pull request by samuelfangjw**
Changes
- Layout
- Colour Scheme
- Maximise screen on startup
- Make panes resizable
- Change icon in taskbar
- Abstract out colours so it's easier to change next time
- Change Icon in README, closes #57 
- Change name of jar file
- Add tabs
- Closes #58

### CODE REVIEW

File: src/main/java/seedu/address/ui/UiManager.java [hunk 1]
```diff
@@ -37,14 +37,16 @@ public UiManager(Logic logic) {
     public void start(Stage primaryStage) {
         logger.info("Starting UI...");
 
-        //Set the application icon.
+        //Set the application icon
```
Comment (vevek) [inline src/main/java/seedu/address/ui/UiManager.java]: I think there should be a space between // and Set.

File: src/main/java/seedu/address/ui/UiManager.java [hunk 2]
```diff
@@ -37,14 +37,16 @@ public UiManager(Logic logic) {
...
         primaryStage.getIcons().add(getImage(ICON_APPLICATION));
 
         try {
             mainWindow = new MainWindow(primaryStage, logic);
             mainWindow.show(); //This should be called before creating other UI parts
             mainWindow.fillInnerParts();
 
+            //Maximise window
```
Comment (vevek) [inline src/main/java/seedu/address/ui/UiManager.java]: I think there should be a space between // and Maximise.

File: src/main/resources/view/HelpWindow.fxml [hunk 3]
```diff
@@ -1,39 +1,40 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <?import javafx.geometry.Insets?>
-<?import javafx.scene.Scene?>
 <?import javafx.scene.control.Button?>
 <?import javafx.scene.control.Label?>
 <?import javafx.scene.image.Image?>
 <?import javafx.scene.layout.HBox?>
+<?import javafx.scene.Scene?>
```
Comment (vevek) [inline src/main/resources/view/HelpWindow.fxml:8]: Nice touch.

File: src/main/resources/view/MainWindow.fxml [hunk 4]
```diff
@@ -1,59 +1,85 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
+<?import java.lang.String?>
 <?import java.net.URL?>
```
Comment (vevek) [inline src/main/resources/view/MainWindow.fxml:4]: Add space below.

**Review by vevek — DISMISSED** [review]
LGTM.

**Comment by samuelfangjw** [comment]
> LGTM.

haha thanks but it's still a draft pr! i'll make some edits tmr and change the pr status when I have something decent :)

File: src/main/java/seedu/address/ui/UiManager.java (hunk 1 above)
Comment (samuelfangjw) [reply src/main/java/seedu/address/ui/UiManager.java]: agreed. just realized the original AB3 does not use spaces consistently, might be worth addressing this in a future pr?

File: src/main/resources/view/MainWindow.fxml (hunk 4 above)
Comment (samuelfangjw) [reply src/main/resources/view/MainWindow.fxml:4]: agree with this, will make changes and update the pr

File: src/main/java/seedu/address/ui/UiManager.java (hunk 1 above)
Comment (vevek) [reply src/main/java/seedu/address/ui/UiManager.java]: Yup, I'll add an issue!

File: src/main/resources/view/MainWindow.fxml (hunk 4 above)
Comment (vevek) [reply src/main/resources/view/MainWindow.fxml:4]: Nice

**Comment by vevek** [comment]
@samuelfangjw Looking forward to it hahaha!

File: src/main/java/seedu/address/ui/UiManager.java (hunk 1 above)
Comment (samuelfangjw) [reply src/main/java/seedu/address/ui/UiManager.java]: thanks! @vevek

File: src/main/java/seedu/address/ui/UiManager.java [hunk 5]
```diff
@@ -20,7 +24,7 @@
     public static final String ALERT_DIALOG_PANE_FIELD_ID = "alertDialogPane";
 
     private static final Logger logger = LogsCenter.getLogger(UiManager.class);
-    private static final String ICON_APPLICATION = "/images/address_book_32.png";
+    private static final String ICON_APPLICATION = "/images/icon.png";
```
Comment (vevek) [inline src/main/java/seedu/address/ui/UiManager.java]: colab_icon_32 would be easier to differentiate.  32 being an example size.

File: src/main/resources/view/DarkTheme.css [hunk 6]
```diff
@@ -212,25 +272,26 @@
...
  */
+
 .button {
     -fx-padding: 5 22 5 22;
-    -fx-border-color: #e2e2e2;
+    -fx-border-color: -ACCENT;
     -fx-border-width: 2;
     -fx-background-radius: 0;
-    -fx-background-color: #1d1d1d;
+    -fx-background-color: -PRIMARY_LIGHT;
```
Comment (vevek) [inline src/main/resources/view/DarkTheme.css]: Perhaps all colour constants can have a prefix. COLOUR_PRIMARY_LIGHT and COLOUR_ACCENT.

File: src/main/resources/view/DarkTheme.css [hunk 7]
```diff
@@ -308,19 +369,19 @@
 }
 
 #cardPane {
-    -fx-background-color: transparent;
+    -fx-background-color: -CELL;
```
Comment (vevek) [inline src/main/resources/view/DarkTheme.css]: COLOUR_CELL would be better. "Cell" is unclear.

File: src/main/resources/view/MainWindow.fxml [hunk 8]
```diff
@@ -1,59 +1,102 @@
...
+            <SplitPane dividerPositions="0.15" orientation="VERTICAL" prefHeight="200.0" prefWidth="160.0">
+              <items>
 
-        <StackPane VBox.vgrow="NEVER" fx:id="resultDisplayPlaceholder" styleClass="pane-with-border"
-                   minHeight="100" prefHeight="100" maxHeight="100">
-          <padding>
-            <Insets top="5" right="10" bottom="5" left="10" />
-          </padding>
-        </StackPane>
+                <StackPane fx:id="resultDisplayPlaceholder" minHeight="NaN" minWidth="NaN"
```
Comment (vevek) [inline src/main/resources/view/MainWindow.fxml]: What's the reason why it is NaN?

**Review by vevek — CHANGES_REQUESTED** [review]
Just some comments. Overall looks fine.

File: src/main/resources/view/MainWindow.fxml (hunk 8 above)
Comment (samuelfangjw) [reply src/main/resources/view/MainWindow.fxml]: I couldn't get the StackPane to have 0 min height with other methods but you're right, it looks a bit odd here. Let me look into this properly.

File: src/main/resources/view/DarkTheme.css (hunk 6 above)
Comment (samuelfangjw) [reply src/main/resources/view/DarkTheme.css]: agreed

File: src/main/java/seedu/address/ui/UiManager.java (hunk 5 above)
Comment (samuelfangjw) [reply src/main/java/seedu/address/ui/UiManager.java]: agreed, will check the other images and relabel them accordingly too.

File: src/main/resources/view/DarkTheme.css (hunk 6 above)
Comment (samuelfangjw) [reply src/main/resources/view/DarkTheme.css]: Using the american spelling color instead for consistency

File: src/main/resources/view/MainWindow.fxml (hunk 8 above)
Comment (samuelfangjw) [reply src/main/resources/view/MainWindow.fxml]: resolved by setting minWidth/Height as necessary. This stops the panes from fully "disappearing" when resizing panes.

**Comment by samuelfangjw** [comment]
latest commit fixes #58

File: src/main/resources/view/ResultDisplay.fxml [hunk 9]
```diff
@@ -3,7 +3,6 @@
 <?import javafx.scene.control.TextArea?>
 <?import javafx.scene.layout.StackPane?>
 
-<StackPane fx:id="placeHolder" styleClass="pane-with-border" xmlns="http://javafx.com/javafx/8"
-    xmlns:fx="http://javafx.com/fxml/1">
-  <TextArea fx:id="resultDisplay" editable="false" styleClass="result-display"/>
+<StackPane fx:id="placeHolder" xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1">
+  <TextArea fx:id="resultDisplay" editable="false" styleClass="result-display" wrapText="true" />
```
Comment (lirc572) [inline src/main/resources/view/ResultDisplay.fxml]: extra space before `/>`

**Review by lirc572 — CHANGES_REQUESTED** [review]
One extra space in fxml. Everything else looks good!

File: src/main/resources/view/MainWindow.fxml (hunk 8 above)
Comment (vevek) [reply src/main/resources/view/MainWindow.fxml]: Nice!

File: src/main/java/seedu/address/commons/core/GuiSettings.java [hunk 10]
```diff
@@ -10,8 +10,8 @@
  */
 public class GuiSettings implements Serializable {
 
-    private static final double DEFAULT_HEIGHT = 600;
-    private static final double DEFAULT_WIDTH = 740;
+    private static final double DEFAULT_HEIGHT = 900;
```
Comment (vevek) [inline src/main/java/seedu/address/commons/core/GuiSettings.java]: Any reason for this specific resolution? Perhaps a generic 1280x720 or 1920x1080 would be consistent among different computers.

**Review by vevek — DISMISSED** [review]
Other than the default resolution and @lirc572 comment, LGTM! Should merge soon to avoid future conflicts.

File: src/main/java/seedu/address/commons/core/GuiSettings.java (hunk 10 above)
Comment (samuelfangjw) [reply src/main/java/seedu/address/commons/core/GuiSettings.java]: Current resolution updated to 1280x720 for better compatibility

**Review by vevek — APPROVED** [review]
LGTM!

**Pull request merged**